kodilla-patterns2/src/main/java/com/kodilla/patterns2/adapter/bookclasifier/libraryb/Statistics.java

```diff
@@ -0,0 +1,37 @@
+package com.kodilla.patterns2.adapter.bookclasifier.libraryb;
+
+import java.util.Arrays;
+import java.util.Map;
+
+public class Statistics implements BookStatistics {
+
+    @Override
+    public int averagePublicationYear(Map<BookSignature, Book> books) {
+        if (books.size() == 0) return 0;
+        int sum = 0;
+        for (Map.Entry<BookSignature, Book> entry : books.entrySet()) {
+            sum += entry.getValue().getYearOfPublication();
+        }
+        return sum / books.size();
+    }
+
+    @Override
+    public int medianPublicationYear(Map<BookSignature, Book> books) {
+        if (books.size() == 0) return 0;
+        int[] years = new int[books.size()];
```

Comment:
Nie musisz korzystać z typu tablicowego. Podobne operacje możesz z powodzeniem wykonać na kolekcjach Javy, wydaje mi się, że kod się odrobinę uprości.
https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#sort-java.util.List-
https://mkyong.com/java8/java-8-how-to-sort-list-with-stream-sorted/